mpisppy/utils/vanilla.py

```diff
@@ -287,6 +289,15 @@ def xhatshuffle_spoke(
         "dump_prefix": "delme",
         "csvname": "looper.csv",
     }
+    if _hasit(args,"add_reversed_shuffle"):
+        xhat_options["xhat_looper_options"]["reverse"] = args.add_reversed_shuffle
+    if _hasit(args,"add_reversed_shuffle"):
+        xhat_options["xhat_looper_options"]["xhatshuffle_iter_step"] = args.xhatshuffle_iter_step
+    
+    if branching_factors:
+        xhat_options["branching_factors"] = branching_factors
+        if all_nodenames is None:
+            all_nodenames = sputils.create_nodenames_from_BFs(branching_factors)
```

Comment:
@bknueven This is related to your last command. If a branching factor is specified without all_nodenames, the spoke creates the all_nodenames

Reply:
Ah, that makes good sense